gateway-server/src/main/java/org/apache/knox/gateway/webshell/WebShellTokenValidator.java

```diff
@@ -0,0 +1,241 @@
+    private static final String JWT_DEFAULT_SIGALG = "RS256";
+    private static final String JWT_EXPECTED_SIGALG = "jwt.expected.sigalg";
+    private static final JWTMessages log = MessagesFactory.get(JWTMessages.class);
+
+    private Map<String,String> params;
+    private String cookieName;
+    private String expectedIssuer;
+    private String expectedSigAlg;
+    private TokenStateService tokenStateService;
+    private JWTokenAuthority authorityService;
+    private SignatureVerificationCache signatureVerificationCache;
+    private JWT token;
+
+    private final GatewayConfig gatewayConfig;
+    private final GatewayServices gatewayServices;
+
+    WebShellTokenValidator(ServletUpgradeRequest req, GatewayServices gatewayServices,
+                           GatewayConfig gatewayConfig){
+        this.gatewayConfig = gatewayConfig;
+        this.gatewayServices = gatewayServices;
+        this.token = getToken(req);
+        if (this.token == null){
+            throw new RuntimeException("no Valid JWT Token");
+        }
+        configureParameters();
+    }
+
+    // todo: call this function again if topology is reloaded
+    private void configureParameters() {
+        params = new LinkedHashMap<>();
+        TopologyService ts = gatewayServices.getService(ServiceType.TOPOLOGY_SERVICE);
+        for (Topology topology : ts.getTopologies()) {
+            if (topology.getName().equals("knoxsso")) {
+                for (Service service : topology.getServices()) {
+                    if (service.getRole().equals("KNOXSSO")) {
+                        params = service.getParams();
+                    }
+                    break;
+                }
+                break;
+            }
+        }
+        cookieName = params.get(KNOXSSO_COOKIE_NAME);
+        if (cookieName == null) {
+            cookieName = DEFAULT_SSO_COOKIE_NAME;
+        }
+        expectedIssuer =  params.get(JWT_EXPECTED_ISSUER);
+        if (expectedIssuer == null) {
+            expectedIssuer = JWT_DEFAULT_ISSUER;
+        }
+        expectedSigAlg =  params.get(JWT_EXPECTED_SIGALG);
+        if (expectedSigAlg == null) {
+            expectedSigAlg = JWT_DEFAULT_SIGALG;
+        }
+        authorityService = gatewayServices.getService(ServiceType.TOKEN_SERVICE);
+        if (isServerManagedTokenStateEnabled()) {
+            tokenStateService = gatewayServices.getService(ServiceType.TOKEN_STATE_SERVICE);
+        }
+        // Setup the verified tokens cache
+        signatureVerificationCache = SignatureVerificationCache.getInstance(
+                "knoxsso", new WebShellFilterConfig(params));
+    }
+
+    private JWT getToken(ServletUpgradeRequest req){
+        List<HttpCookie> ssoCookies = req.getCookies();
+        for (HttpCookie ssoCookie : ssoCookies) {
+            if (cookieName.equals(ssoCookie.getName())) {
+                try {
+                    token = new JWTToken(ssoCookie.getValue());
```

Comment:
I think this is confusing. This method both overwrites the `token` instance variable and returns a value. I would only keep the return value and to the overwrite outside.